### PR TITLE
Support Futures

### LINKED ISSUE #86: Native support for Tornado futures
Currently, if a method being doubled is an asynchronous operation, we use `tornado.gen.maybe_future` to wrap values into futures, and for exceptions for asynchronous operations, we use this:

    def failed_future(exc):
        future = tornado.gen.Future()
        future.set_exception(exc)
        return future

It would be great if doubles supported futures natively. The API could possibly look like,

    allow(foo).bar.and_raise.async(Exception('foo'))
    allow(foo).bar.and_return.async(result)

Doubles won't have to depend on Tornado for this and can just use the futures from `concurrent.futures` since Tornado supports those too.

### PR DESCRIPTION
**Pull request by toddsifleet**
* Allow stubbing methods with `and_return_future` and `and_raise_future`
* Resolves #86

### CODE REVIEW

**Comment by blampe** [comment]
changelog

File: doubles/allowance.py [hunk 1]
```diff
@@ -1,6 +1,8 @@
 from functools import wraps
 from inspect import getargspec
 
+from concurrent.futures import Future
```
Comment (lucasmarshall) [inline doubles/allowance.py]: This import should probably go inside the functions below so we don't break doubles entirely when futures is are not supported.

**Comment by abhinav** [comment]
LGTM :shipit:

**Comment by blampe** [comment]
+1

File: doubles/allowance.py [hunk 2]
```diff
@@ -9,6 +9,19 @@
...
 
 
+def _get_future():
+    try:
+        from concurrent.futures import Future
+    except ImportError:
+        try:
+            from tornado.concurrent import Future
+        except ImportError:
+            raise ImportError(
```
Comment (breerly) [inline doubles/allowance.py]: nice

**Comment by breerly** [comment]
:ship: :it:

**Comment by breerly** [comment]
Please make sure this is a minor version release, `1.1.0` - we shouldn't be adding features to patch-level releases. semver.org

**Comment by toddsifleet** [comment]
This was merged in.

**Pull request closed without merging**